framework/src/st_things/things_stack/easy-setup/easysetup_manager.c

```diff
@@ -168,7 +168,7 @@ int esm_set_device_property_by_app(char *name, const wifi_mode_e *mode, int ea_m
 
 	int i = 0;
 
-	if (mode == NULL || ea_mode < 1 || freq >= WiFi_FREQ_EOF) {
+	if (mode == NULL || ea_mode < 1 | freq < WiFi_24G || WiFi_FREQ_EOF <= freq) {
```

Comment:
I think readabilty is not good.
```var < XX || var > YY``` is better I think.

Reply:
OK. i chaged